### PR TITLE
Beginner sitrep hints

### PR DESCRIPTION
**Pull request by dbenage-cx**
Display beginner hints as a sitep in a semi-random fashion.

From discussion [on forums](http://www.freeorion.org/forum/viewtopic.php?f=6&t=9955)

Adds a focus article to 'pedia, utilized autofactories icon as placeholder.

### CODE REVIEW

File: default/stringtables/en.txt [hunk 1]
```diff
@@ -3918,6 +3918,15 @@ Outpost
 OUTPOSTS_TEXT
 Outposts are unmanned stations, which can be founded in places too hostile for a colony to survive. They have no population, and normally produce no resources. But they create [[encyclopedia SUPPLY_TITLE]] (with the right tech) and provide vision. Colonies can be founded on planets that already have an Outpost. To build an Outpost, you need to put an [[shippart CO_OUTPOST_POD]] on a ship. Outposts are also formed when the population of a colony drops to zero for any reason.
 
+FOCUS_TITLE
+Planetary Focus
+FOCUS_TEXT
+'''Each planet can be focused on a specific area, such as production or research.  Not all areas of focus are always available.  They may be restricted because the species can not focus on that area, the technology is not yet known, or other factors such as the availability of a special.
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: wording "focused on a specific area" is awkward.

File: default/stringtables/en.txt [hunk 2]
```diff
@@ -3918,6 +3918,15 @@ Outpost
...
+
+The focus of a planet may grant it bonuses or abilities, depending on the focus.  Changing a planets focus occurs a small penalty that will diminish over time.
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: "abilities" seems like odd word choice... "effects" might be better.

"The focus ... the focus." is awkward.

File: default/stringtables/en.txt [hunk 3]
```diff
@@ -3918,6 +3918,15 @@ Outpost
...
+
+A planet will default to the inhabiting species Preferred Focus if it has one, otherwise it will default to production.'''
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: species'

why capitals for "Preferred Focus"?

File: default/stringtables/en.txt [hunk 4]
```diff
@@ -5047,6 +5056,69 @@ SITREP_EMPIRE_ELIMINATED_LABEL
...
+####           RANDOM BEGINNER HINTS                     ####
+#############################################################
+BEGINNER_HINTS
+Random Game Hints
+
+RANDOM_BEGINNER_HINT
+%rawtext%
+
+BEGINNER_HINT_01
+A good first step is to build scout ships and explore the region around your home system.
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: first step when?

File: default/stringtables/en.txt [hunk 5]
```diff
@@ -5047,6 +5056,69 @@ SITREP_EMPIRE_ELIMINATED_LABEL
...
+If you can conquer planets with other species and add them to your empire, it may be easier for them to colonize different environments.  Not all native species can colonize and some can not build ships.
+
+BEGINNER_HINT_03
+Ships will be repaired in a single turn at a system that has a friendly [[buildingtype BLD_SHIPYARD_ORBITAL_DRYDOCK]] if there are no enemies in the system that turn.
+
+BEGINNER_HINT_04
+Systems that are connected by thick lines are [[encyclopedia SUPPLY_TITLE]] line linked and they share [[encyclopedia INDUSTRY_TITLE]] Production Points (PP).
+
+BEGINNER_HINT_05
+All system share [[encyclopedia RESEARCH_TITLE]] Points (RP) that they generate even if they are not connected by supply lines. If you have a remote system that is unconnected, setting the [[encyclopedia FOCUS_TITLE]] to research might be a more efficient use.
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: more efficient than what?

File: default/stringtables/en.txt [hunk 6]
```diff
@@ -5047,6 +5056,69 @@ SITREP_EMPIRE_ELIMINATED_LABEL
...
+Random Game Hints
+
+RANDOM_BEGINNER_HINT
+%rawtext%
+
+BEGINNER_HINT_01
+A good first step is to build scout ships and explore the region around your home system.
+
+BEGINNER_HINT_02
+If you can conquer planets with other species and add them to your empire, it may be easier for them to colonize different environments.  Not all native species can colonize and some can not build ships.
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: awkward wording.

File: default/stringtables/en.txt [hunk 7]
```diff
@@ -5047,6 +5056,69 @@ SITREP_EMPIRE_ELIMINATED_LABEL
...
+Ships will be repaired in a single turn at a system that has a friendly [[buildingtype BLD_SHIPYARD_ORBITAL_DRYDOCK]] if there are no enemies in the system that turn.
+
+BEGINNER_HINT_04
+Systems that are connected by thick lines are [[encyclopedia SUPPLY_TITLE]] line linked and they share [[encyclopedia INDUSTRY_TITLE]] Production Points (PP).
+
+BEGINNER_HINT_05
+All system share [[encyclopedia RESEARCH_TITLE]] Points (RP) that they generate even if they are not connected by supply lines. If you have a remote system that is unconnected, setting the [[encyclopedia FOCUS_TITLE]] to research might be a more efficient use.
+
+BEGINNER_HINT_06
+Ships will refuel when they are in a system that is connected by thick or by thin lines, as long as the thin lines extend back to thick supply line connected systems.
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: rephrase "thin lines"... a bit unclear what it means, and only applicable to an empire's own supply connected systems.

File: default/stringtables/en.txt [hunk 8]
```diff
@@ -5047,6 +5056,69 @@ SITREP_EMPIRE_ELIMINATED_LABEL
...
+
+BEGINNER_HINT_07
+Some buildings, like the [[buildingtype BLD_INDUSTRY_CENTER]], will boost production on all supply connected planets. Others, like [[buildingtype BLD_GAS_GIANT_GEN]], only help in the system that they are built in. Always check the 'Pedia for specifics on different buildings!
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: are there multiple buildings that have these effects?

"Always" check? In what situation? Rephrase to "Check the Pedia for details about the functions of buildings" or similar.

File: default/stringtables/en.txt [hunk 9]
```diff
@@ -5047,6 +5056,69 @@ SITREP_EMPIRE_ELIMINATED_LABEL
...
+
+BEGINNER_HINT_08
+An Outpost ship can setup an Outpost on any unclaimed visible planet or asteroid that it can reach. Outposts can be settled by any species in your empire that has a connection to it, at least 5 happiness, and 3 population.
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: unless there are hostile forces in system...

settled by producing a colony building [conditions], or using a colony ship.

noting that target population matters as well...

File: default/stringtables/en.txt [hunk 10]
```diff
@@ -5047,6 +5056,69 @@ SITREP_EMPIRE_ELIMINATED_LABEL
...
+
+BEGINNER_HINT_09
+Colony ships take longer to build than outpost ships but can settle planets outside of your supply range.
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: comma between ships and but

File: default/stringtables/en.txt [hunk 11]
```diff
@@ -5047,6 +5056,69 @@ SITREP_EMPIRE_ELIMINATED_LABEL
...
+
+BEGINNER_HINT_10
+Right click on a planet to access the Planet Suitability report. This will tell you what species from your empire will be able to successfully settle that planet. Numbers in green show the maximum population that the planet could achieve if colonized by that species, red numbers indicate that the species would die out if they attempted to colonize.
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: right click on a planet where?

File: default/stringtables/en.txt [hunk 12]
```diff
@@ -5047,6 +5056,69 @@ SITREP_EMPIRE_ELIMINATED_LABEL
...
+
+BEGINNER_HINT_11
+Maximum populations that planets can achieve can be boosted by researching various growth techs, or by utilizing various rare specials that can be found on some planets.
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: note relevance of specials for particular metabolism types

File: default/stringtables/en.txt [hunk 13]
```diff
@@ -5047,6 +5056,69 @@ SITREP_EMPIRE_ELIMINATED_LABEL
...
+
+BEGINNER_HINT_12
+Be on the lookout for specials on planets like [[encyclopedia GROWTH_FOCUS_TITLE]]', they usually indicate some special resource on that planet that may benefit your entire empire if utilized properly. Right click on a special's icon to access its 'Pedia entry and find out more!
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: specials like ..., or planets like ... ?

"utilized properly" ?

File: default/stringtables/en.txt [hunk 14]
```diff
@@ -5047,6 +5056,69 @@ SITREP_EMPIRE_ELIMINATED_LABEL
...
+
+BEGINNER_HINT_13
+Only [[species SP_EXOBOT]] can settle asteroids. If you find a lot of asteroids belts nearby you might want to research how to build them sooner than later!
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: asteroid belts, no s on asteroid

sooner rather than later

File: default/stringtables/en.txt [hunk 15]
```diff
@@ -5047,6 +5056,69 @@ SITREP_EMPIRE_ELIMINATED_LABEL
...
+
+BEGINNER_HINT_14
+Research [[encyclopedia SUPPLY_TITLE]] improvements to increase the distance from your systems that your supply line reaches.
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: supply lines reach

File: default/stringtables/en.txt [hunk 16]
```diff
@@ -5047,6 +5056,69 @@ SITREP_EMPIRE_ELIMINATED_LABEL
...
+
+BEGINNER_HINT_16
+You may want to hold off on researching [[tech LRN_PSIONICS]] until your empire includes a species with [[encyclopedia TELEPATHIC_TITLE]].
+
+BEGINNER_HINT_17
+[[encyclopedia SHIELDS_TITLE]] on ships defend against every shot that hits the ship, reducing the damage by the amount of the shield value. Ship shield values can be affected by various factors including space storms, what type of star system you are in, or if you are close to a captured empire capital planet.
+
+BEGINNER_HINT_18
+[[encyclopedia FOCUS_TITLE]] - pay attention to the focus setting of new planets that are added to your empire, either from settling them or from conquering them. This controls if the majority of the planet focuses on industrial production or on research or on defense. Later in the game more options become available as you research new technologies. Not all species are able to focus on [[encyclopedia RESEARCH_TITLE]] or [[encyclopedia INDUSTRY_TITLE]].
```
Comment (geoffthemedio) [inline default/stringtables/en.txt]: "the majority" is misleading / vague

comma after later in the game

more focus options may unlock as new technologies are researched, and different species may have different focus options available.

File: default/stringtables/en.txt (hunk 8 above)
Comment (dbenage-cx) [inline default/stringtables/en.txt]: For production: Black Hole Power Generator and Solar Orbital Generation.  Rewording this from boosting production to having an effect.

File: default/stringtables/en.txt [hunk 17]
```diff
@@ -5077,34 +5077,34 @@ BEGINNER_HINT_04
...
 
 BEGINNER_HINT_11
-Maximum populations that planets can achieve can be boosted by researching various growth techs, or by utilizing various rare specials that can be found on some planets.
+Maximum populations that planets can achieve can be boosted by researching various growth techs.  Rare specials can also increase the maximum population for species with certain metabolisms, see [[encyclopedia GROWTH_FOCUS_TITLE]] for more info on these specials.
 
 BEGINNER_HINT_12
-Be on the lookout for specials on planets like [[encyclopedia GROWTH_FOCUS_TITLE]]', they usually indicate some special resource on that planet that may benefit your entire empire if utilized properly. Right click on a special's icon to access its 'Pedia entry and find out more!
+Be on the lookout for planets with specials, they usually indicate some special resource on that planet that may benefit your entire empire if utilized properly.  Right click on a special's icon to access its 'Pedia entry and find out more!
```
Comment (Morlic-fo) [inline default/stringtables/en.txt]: Double whitespace introduced?

File: default/stringtables/en.txt [hunk 18]
```diff
@@ -5077,34 +5077,34 @@ BEGINNER_HINT_04
...
+Some buildings, like the [[buildingtype BLD_INDUSTRY_CENTER]], will have an effect on each planet that has a supply line connection to it. Others, like [[buildingtype BLD_GAS_GIANT_GEN]], only affect the system that they are built in.  Check the 'Pedia to learn about the functions of different buildings.
 
 BEGINNER_HINT_08
-An Outpost ship can setup an Outpost on any unclaimed visible planet or asteroid that it can reach. Outposts can be settled by any species in your empire that has a connection to it, at least 5 happiness, and 3 population.
+An Outpost ship can setup an outpost on any unclaimed visible planet or asteroid that it can reach, unless there are hostile ships in that system.
 
 BEGINNER_HINT_09
-Colony ships take longer to build than outpost ships but can settle planets outside of your supply range.
+Colony ships take longer to build than outpost ships, but can settle planets outside of your supply range.
```
Comment (Morlic-fo) [inline default/stringtables/en.txt]: Pretty sure the comma shouldn't be there.

File: default/stringtables/en.txt [hunk 19]
```diff
@@ -5077,34 +5077,34 @@ BEGINNER_HINT_04
 Systems that are connected by thick lines are [[encyclopedia SUPPLY_TITLE]] line linked and they share [[encyclopedia INDUSTRY_TITLE]] Production Points (PP).
 
 BEGINNER_HINT_05
-All system share [[encyclopedia RESEARCH_TITLE]] Points (RP) that they generate even if they are not connected by supply lines. If you have a remote system that is unconnected, setting the [[encyclopedia FOCUS_TITLE]] to research might be a more efficient use.
+All system share [[encyclopedia RESEARCH_TITLE]] Points (RP) that they generate even if they are not connected by supply lines.  When a system is not connected, setting the [[encyclopedia FOCUS_TITLE]] to research might be more efficient than other settings.
 
 BEGINNER_HINT_06
-Ships will refuel when they are in a system that is connected by thick or by thin lines, as long as the thin lines extend back to thick supply line connected systems.
+Ships will refuel when they are in a system that is connected to a [[encyclopedia SUPPLY_TITLE]] line.  Narrow supply lines, which are displayed with a thin line between systems, will also refuel ships if they are connected to a wider supply line.
```
Comment (Morlic-fo) [inline default/stringtables/en.txt]: Normal starlanes are also displayed by a thin line between systems. Probably should mention the coloring somewhere.

File: default/stringtables/en.txt [hunk 20]
```diff
@@ -5077,34 +5077,34 @@ BEGINNER_HINT_04
 Systems that are connected by thick lines are [[encyclopedia SUPPLY_TITLE]] line linked and they share [[encyclopedia INDUSTRY_TITLE]] Production Points (PP).
 
 BEGINNER_HINT_05
-All system share [[encyclopedia RESEARCH_TITLE]] Points (RP) that they generate even if they are not connected by supply lines. If you have a remote system that is unconnected, setting the [[encyclopedia FOCUS_TITLE]] to research might be a more efficient use.
+All system share [[encyclopedia RESEARCH_TITLE]] Points (RP) that they generate even if they are not connected by supply lines.  When a system is not connected, setting the [[encyclopedia FOCUS_TITLE]] to research might be more efficient than other settings.
```
Comment (Morlic-fo) [inline default/stringtables/en.txt]: "not connected" should mention supply somewhere.

File: default/stringtables/en.txt (hunk 17 above)
Comment (dbenage-cx) [inline default/stringtables/en.txt]: Double space at start of sentence.  Will go through and correct any missed.

File: default/stringtables/en.txt (hunk 18 above)
Comment (dbenage-cx) [inline default/stringtables/en.txt]: https://github.com/freeorion/freeorion/pull/526#discussion_r56298545
I don't see a comma as necessary, but think it reads fine either way.

Comment (Morlic-fo) [inline default/stringtables/en.txt]: http://www.grammarly.com/handbook/punctuation/comma/23/comma-before-but/

`Colony ships take longer to build than outpost ships but can settle planets outside of your supply range.` --> ok
`Colony ships take longer to build than outpost ships, but they can settle planets outside of your supply range.` --> ok
`Colony ships take longer to build than outpost ships, but can settle planets outside of your supply range.` --> not ok.

File: default/stringtables/en.txt (hunk 19 above)
Comment (dbenage-cx) [inline default/stringtables/en.txt]: As these details are displayed in a more readable manner in the linked 'pedia entry, could this just be reduced to:
'[[encyclopedia SUPPLY_TITLE]] lines can be used to refill the [[encyclopedia FUEL_TITLE]] supply of ships.'

File: default/customizations/custom_sitreps.txt [hunk 21]
```diff
@@ -190,9 +190,41 @@ EffectsGroup
 //             ]
 //             empire = Source.Owner
 
+//#####     RANDOM BEGINNER HINTS     #####//
 
-
-
+EffectsGroup
+    scope = Source
+    activation = Turn low = 0 high = 200
```
Comment (geoffthemedio) [inline default/customizations/custom_sitreps.txt]: Just notice this... should it really be shown every turn, especially if there are only 19 hits?

Comment (dbenage-cx) [inline default/customizations/custom_sitreps.txt]: entries will repeat before the list is completely iterated.  The threshold could be lowered to turn 100 or so, or a different selection method used, though I think the hints should be once per turn (or in this case, an attempt once per turn).

**Comment by geoffthemedio** [comment]
I feel like the hints should also have a unique icon, and something at the start of the text indicating that they are a hint, like "HINT #6: Cray are dumb"

**Comment by dbenage-cx** [comment]
Re-branded 'pedia icon.  Probably just a placeholder until inspiration strikes.  All that comes to mind currently are either a question mark or exclamation point.

**Comment by geoffthemedio** [comment]
Can you get rid of double-spaces after periods, and then squash into one commit?

**Comment by dbenage-cx** [comment]
squashed

**Pull request merged**